models.py

```diff
@@ -28,12 +32,42 @@ def generate_compiled_segmentation_model(model_name, model_parameters, num_class
                                          optimizing_class_id=None, optimizing_input_threshold=None,
                                          optimized_class_thresholds=None):
 
-    # These are the only model, loss, and optimizer currently supported
-    assert model_name == 'Unet'
-    assert loss == 'cross_entropy'
-    assert optimizer == 'adam'
+    # alter input_shape due to inability of yaml to accept tupples!
+    if 'input_shape' in model_parameters:
+        model_parameters['input_shape'] = tuple(model_parameters['input_shape'])
+    else:  # to guarantee the old config files still work
+        model_parameters['input_shape'] = (None, None, 1)
+
+    # These are the only optimizer currently supported
+    if optimizer.lower() == 'adam':
+        opt_fn = Adam()
+    elif optimizer.lower() == 'sdg':
+        opt_fn = SGD()
+    elif optimizer.lower() == 'adagrad':
+        opt_fn = Adagrad()
+    elif optimizer.lower() == 'adamax':
+        opt_fn = Adamax()
+    elif optimizer.lower() == 'ftrl':
+        opt_fn = Ftrl()
+    elif optimizer.lower() == 'nadam':
+        opt_fn = Nadam()
+    elif optimizer.lower() == 'rmsprop':
+        opt_fn = RMSprop()
+    else:
+        raise NameError("Optimizer not supported")
+
+    # These are the only loss currently supported
+    if loss == 'cross_entropy':
```

Comment:
lets differentiate clearly now. 'binary_cross_entropy' here

Reply:
but this means that the V1 config files will not work. can we leave as "binary_cross_entropy" or "cross_entropy"? @rak5216

Reply:
used "or" statement